src/main/java/tutu/data/TaskList.java

```diff
@@ -0,0 +1,116 @@
+package tutu.data;
+
+import java.util.ArrayList;
+import tutu.task.Task;
+import tutu.exception.InvalidInputException;
+
+/** Represents a TaskList object to store the tasks. */
+public class TaskList {
+    /** Stores the tasks added as Task objects in an ArrayList object. */
+    private ArrayList<Task> items;
+
+    /**
+     * Constructor to create a TaskList Object.
+     */
+    public TaskList() {
+        this.items = new ArrayList<Task>();
+    }
+
+    /**
+     * Returns the number of tasks in the TaskList object.
+     * @return Number of tasks.
+     */
+    public int taskNumber() {
+        return items.size();
+    }
+
+    /**
+     * Retrieves the task at a specified index.
+     * @param i Index of task to be retrieved.
+     * @return Task object representing the task at index i.
+     */
```

Comment:
Perhaps adding an empty line between description and parameter section is better according to Java coding standard? I noticed this issue in several other places too.
```suggestion
    /**
     * Retrieves the task at a specified index.
     *
     * @param i Index of task to be retrieved.
     * @return Task object representing the task at index i.
     */
```